tests/pruners_tests/test_median.py

```diff
@@ -120,16 +122,18 @@ def test_median_pruner_interval_steps(
     study = optuna.study.create_study()
 
     trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
-    n_steps = max(expected_prune_steps)
-    base_index = 0
-    for i in range(base_index, base_index + n_steps):
-        trial.report(base_index, i)
+    last_step = max(expected_prune_steps) + 1
+
+    for i in range(last_step):
+        trial.report(0, i)
     study._storage.set_trial_state(trial._trial_id, TrialState.COMPLETE)
 
     trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
```

Comment:
Repeat my comment above:

If I understand correctly, we can use `optuna.trial.create_trial` here, and I think we should use it instead as per the disclaimer of https://optuna.readthedocs.io/en/stable/reference/generated/optuna.trial.Trial.html#optuna.trial.Trial (quote follows).

> Note that the direct use of this constructor is not recommended. This object is seamlessly instantiated and passed to the objective function behind the optuna.study.Study.optimize() method; hence library users do not care about instantiation of this object.

This opinion is based on my take that we should not use some API if it is discouraged or deprecated.
So, what's your take?